DOCSP-35201 Transaction Retry Logic in Core API (#1164)
mongodb/docs-node: source/crud/transactions.txt

Transaction Errors
------------------

> Since MongoDB transactions are :website:`ACID compliant

Reviewer: S: Add a Convenient Transaction API subheading to distinguish from Core API, similar to the Transaction APIs section earlier on the page. The list of errors can go before the subheadings.

</basics/acid-transactions>`, the driver may produce errors during operation to
ensure your data maintains consistent. If the following errors occur, the

> transaction must be retried:

Reviewer: I: [second person imperative](https://www.mongodb.com/docs/meta/style-guide/writing/write-to-the-user/). It looks like the driver auto-retries with one API but the user needs to manually retry with the other? To cover both cases, you could change this to something like 'The following sections describe how to handle these errors for different APIs', as an intro to the subheadings.

- ``TransientTransactionError``: Raised if a write operation encounters an error
  before the driver commits the transaction. To learn more about this error
  type, see the :manual:`TransientTransactionError
  description </core/transactions-in-applications/#transienttransactionerror>` on
  the Drivers API page in the Server manual.
- ``UnknownTransactionCommitResult``: Raised if the commit operation encounters
  an error. To learn more about this error type, see the
  :manual:`UnknownTransactionCommitResult
  description </core/transactions-in-applications/#unknowntransactioncommitresult>`
  on the Drivers API page in the Server manual.

The Convenient Transaction API incorporates retry logic for these error types,
so the driver automatically retries the transaction until there is a successful
commit.

Core API Error Handling
~~~~~~~~~~~~~~~~~~~~~~~

> If you are using the Core API to perform a transaction, you must implement the
> error-handling logic into your application. To do so, you must create the
> following functions:

Reviewer: S:

```suggestion
If you are using the Core API to perform a transaction, you must add the
following error-handling functions to your application:
```
